velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -655,12 +653,18 @@ TypePtr ReaderBase::convertType(
         return DOUBLE();
       case thrift::Type::type::BYTE_ARRAY:
       case thrift::Type::type::FIXED_LEN_BYTE_ARRAY:
-        if (binaryAsString) {
-          return VARCHAR();
+        if (options_.fileSchema() != nullPtr 
```

Comment:
You probably need to add `requestedType` to `getParquetColumnInfo` to handle nested `VARBINARY` as well